app/assets/src/components/BulkUploadImport.jsx

```diff
@@ -497,6 +502,29 @@ class BulkUploadImport extends React.Component {
     });
   }
 
+  // Update sample names if they already exist in the project
+  resolveSampleNames = async () => {
+    const updatedSamples = this.state.samples.map(async sample => {
+      // For each sample, ask the server for a validated name
+      const resp = await postWithCSRF(
```

Comment:
Can you move `validateSampleName` into `api/index.js` as a function?

Also, in the new sample upload flow, you can only upload to one project at a time. I'll probably end up refactoring this to check all the sample names in batch (though I guess it doesn't make sense to do it right now).

Reply:
👍